### PR TITLE
add match_mode example

### PR DESCRIPTION
**Pull request by MatFi**
as suggested by @c42f in #25612

### CODE REVIEW

File: stdlib/Test/src/logging.jl [hunk 1]
```diff
@@ -136,6 +136,11 @@ If we also wanted to test the debug messages, these need to be enabled with the
 
     @test_logs (:info,"Doing foo with n=2") (:debug,"Iteration 1") (:debug,"Iteration 2") min_level=Debug foo(2)
 
+In case the logs get huge probably we want to test if only a particular set of
+messages is contained, here the `match_mode` keyword becomes useful:
```
Comment (c42f) [inline stdlib/Test/src/logging.jl]: How about:

> If you want to test that some particular messages are generated while ignoring the rest, you can set the keyword  `match_mode=:any`:

**Review by c42f — COMMENTED** [review]
Great, thanks! I suggest a little rewording for clarity, but other than that this looks good to me.

File: stdlib/Test/src/logging.jl (hunk 1 above)
Comment (MatFi) [reply stdlib/Test/src/logging.jl]: sounds good.

**Comment by c42f** [comment]
CI failures look unrelated (OSX packaging problem / WIN32 backtrace error)

**Pull request merged**